tfhe/src/integer/gpu/ciphertext/boolean_value.rs

```diff
@@ -188,3 +190,9 @@ impl AsMut<CudaUnsignedRadixCiphertext> for CudaBooleanBlock {
         &mut self.0
     }
 }
+impl Clone for CudaBooleanBlock {
+    fn clone(&self) -> Self {
+        let streams = CudaStreams::new_multi_gpu();
+        Self(self.0.duplicate(&streams))
+    }
+}
```

Comment:
No sure we should impl that, here as CudaBooleanBlock in as integer thing and, CudaUnsined/SignedRadix do not impl Clone

Maybe we have to the same thing as the integer backend  (have a CudaOwned(Vec<..>) and impl Clone for it

Or move the `impl Clone for CudaBooleanBlock` to integer (and  probably impl it also for Radix types)

Reply:
Ok I'll follow the same logic with GpuOwned, etc for Booleans.

Reply:
I introduced GpuBooleanSlice, GpuBooleanSliceMut, GpuBooleanOwned to do it, the change is pushed.